colearn_grpc/example_grpc_learner_client.py

```diff
@@ -0,0 +1,237 @@
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2021 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+import time
+from concurrent.futures import ThreadPoolExecutor
+from queue import Queue, Empty
+from threading import Lock
+
+import grpc
+from google.protobuf import empty_pb2
+from prometheus_client import Summary
+
+import colearn_grpc.proto.generated.interface_pb2 as ipb2
+import colearn_grpc.proto.generated.interface_pb2_grpc as ipb2_grpc
+from colearn.ml_interface import MachineLearningInterface, ProposedWeights, Weights
+from colearn_grpc.logging import get_logger
+from colearn_grpc.utils import iterator_to_weights, weights_to_iterator
+
+_logger = get_logger(__name__)
+
+_time_query = Summary("contract_learner_grpc_client_query_time",
+                      "Metric measures the time it takes to interrogate ML side about supported models")
+_time_setup = Summary("contract_learner_grpc_client_setup_time",
+                      "Metric measures the time it takes to initialize the ML side with a selected model")
+_time_train = Summary("contract_learner_grpc_client_train_time",
+                      "Metric measures the time it takes to train the selected model")
+_time_test = Summary("contract_learner_grpc_client_test_time",
+                     "Metric measures the time it takes to test")
+_time_accept = Summary("contract_learner_grpc_client_accept_time",
+                       "Metric measures the time it takes to accept weights")
+_time_get = Summary("contract_learner_grpc_client_get_time",
+                    "Metric measures the time it takes to get the current weights")
+
+
+class GRPCLearnerClient(MachineLearningInterface):
```

Comment:
We need a simple gRPC client in colearn for demonstration purposes. This one is mostly copied from the full client in contract_learn; suggestions for what to simplify would be welcome!